src/providers/invoice/invoice.ts

```diff
@@ -0,0 +1,99 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { ImageLoader } from 'ionic-image-loader';
+import { ConfigProvider } from '../config/config';
+import { EmailNotificationsProvider } from '../email-notifications/email-notifications';
+import { GiftCardProvider } from '../gift-card/gift-card';
+import { HomeIntegrationsProvider } from '../home-integrations/home-integrations';
+import { Logger } from '../logger/logger';
+import { PersistenceProvider } from '../persistence/persistence';
+import { TimeProvider } from '../time/time';
+
+@Injectable()
+export class InvoiceProvider extends GiftCardProvider {
+  constructor(
+    public configProvider: ConfigProvider,
+    public imageLoader: ImageLoader,
+    public emailNotificationsProvider: EmailNotificationsProvider,
+    public http: HttpClient,
+    public logger: Logger,
+    public homeIntegrationsProvider: HomeIntegrationsProvider,
+    public persistenceProvider: PersistenceProvider,
+    public timeProvider: TimeProvider
+  ) {
+    super(
+      configProvider,
+      emailNotificationsProvider,
+      http,
+      imageLoader,
+      logger,
+      homeIntegrationsProvider,
+      persistenceProvider,
+      timeProvider
+    );
+    this.logger.debug('InvoiceProvider initialized');
+    this.setCredentials();
+  }
+
+  public async getBitPayInvoiceData(id: string) {
```

Comment:
Gift Card getBitPayInvoice uses /invoices end point. It does not give merchant name or buyer email so I made a new method to use /invoiceData end point